examples/discovery-search-app/cypress/integration/component_tests/search_input/spec.ts

```diff
@@ -0,0 +1,136 @@
+describe('SearchInput component', () => {
+  beforeEach(() => {
+    cy.server();
+    cy.visit('/');
+  });
+
+  describe('When the SearchInput loads', () => {
+    it('has placeholder text "Search"', () => {
+      cy.get('.bx--search-input').should('have.attr', 'placeholder', 'Search');
+    });
+
+    it('has magnifying glass icon', () => {
+      cy.get('.bx--search-magnifier').should('be.visible');
+    });
+  });
+
+  // Basic query tests
+  describe('When entering a query', () => {
+    beforeEach(() => {
+      cy.fixture('components/search_input/query.json').as('queryJSON');
```

Comment:
for directory structure, what if we focused more on use-cases rather than individual components? something more like
- `spelling`
- `autocompletion`
- `pagination`
- `dynamic_facets`
- `collection_filter`
- `single_select`
- `multi_select_facets`
- `table_results`
- `passage_results`

Reply:
I can get behind that approach. I started organizing these on a component-level because that's how Laxmi's sets of tests were laid out, but I think use-cases makes more sense as a foundation. I already started running into fuzzy areas that this would fix.